src/CMakeLists.txt

```diff
@@ -20,20 +20,24 @@ if (USE_CONAN)
     # Download automatically, you can also just copy the conan.cmake file
     if (NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
         message(STATUS "Downloading conan.cmake from https://github.com/conan-io/cmake-conan")
-        file(DOWNLOAD "https://github.com/conan-io/cmake-conan/raw/v0.15/conan.cmake"
+        file(DOWNLOAD "https://github.com/conan-io/cmake-conan/raw/v0.16.1/conan.cmake"
             "${CMAKE_BINARY_DIR}/conan.cmake")
     endif ()
 
     include(${CMAKE_BINARY_DIR}/conan.cmake)
     #conan_add_remote(NAME bincrafters INDEX 1 URL https://api.bintray.com/conan/bincrafters/public-conan)
     conan_add_remote(NAME jfrog INDEX 1 URL https://turtlebrowser.jfrog.io/artifactory/api/conan/conan)
 
+    conan_cmake_autodetect(settings)
+
     if (UPDATE_CONAN)
-        conan_cmake_run(CONANFILE ../conanfile.py BASIC_SETUP CMAKE_TARGETS UPDATE BUILD outdated)
+        conan_cmake_install(PATH_OR_REFERENCE ${CMAKE_CURRENT_SOURCE_DIR}/../conanfile.py UPDATE BUILD outdated SETTINGS ${settings})
     else ()
-        conan_cmake_run(CONANFILE ../conanfile.py BASIC_SETUP CMAKE_TARGETS BUILD missing)
+        conan_cmake_install(PATH_OR_REFERENCE ${CMAKE_CURRENT_SOURCE_DIR}/../conanfile.py BUILD missing SETTINGS ${settings})
```

Comment:
```suggestion
        conan_cmake_install(PATH_OR_REFERENCE ${CMAKE_CURRENT_SOURCE_DIR}/../conanfile.py BUILD missing SETTINGS ${settings} SETTINGS compiler.cppstd=17)
```

Reply:
Looks like libmysqlclient doesn't build with this on mac at least

Reply:
I reverted it